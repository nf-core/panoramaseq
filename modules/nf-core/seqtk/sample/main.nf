process SEQTK_SAMPLE {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container 'biocontainers/seqtk:1.4--he4a0461_1'

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*.fastq.gz"), emit: reads
    path "versions.yml"                , emit: versions

    when:
    meta.sample_size != null && (task.ext.when == null || task.ext.when)

    script:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def sample_size = meta.sample_size
    if (!(args ==~ /.*\ -s\ ?[0-9]+.*/)) {
        args += " -s100"
    }
    if (!sample_size) {
        error "SEQTK/SAMPLE must have a sample_size value included in meta"
    }
    """
    printf "%s\\n" $reads | while read f;
    do
        seqtk \\
            sample \\
            $args \\
            \$f \\
            $sample_size \\
            | gzip --no-name > ${prefix}_\$(basename \$f)
    done

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqtk: \$(echo \$(seqtk 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    echo "" | gzip > ${prefix}.fastq.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqtk: \$(echo \$(seqtk 2>&1) | sed 's/^.*Version: //; s/ .*\$//')
    END_VERSIONS
    """

}
