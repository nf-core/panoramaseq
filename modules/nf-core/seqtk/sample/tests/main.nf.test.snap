{
    "sarscov2_sample_singlend_fqgz_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true,
                            "sample_size": 50
                        },
                        "test.sampled.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0529f2d163df9e2cd2ae8254dfb63806"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true,
                            "sample_size": 50
                        },
                        "test.sampled.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0529f2d163df9e2cd2ae8254dfb63806"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-20T12:54:32.569666487"
    },
    "sarscov2_sample_pairedend_fqgz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sample_size": 50
                        },
                        [
                            "test.sampled_test_1.fastq.gz:md5,e5f44fafd7351c5abb9925a075132941",
                            "test.sampled_test_2.fastq.gz:md5,b8e8bdabeac41e066052d1ea3abcf68e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,0529f2d163df9e2cd2ae8254dfb63806"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "sample_size": 50
                        },
                        [
                            "test.sampled_test_1.fastq.gz:md5,e5f44fafd7351c5abb9925a075132941",
                            "test.sampled_test_2.fastq.gz:md5,b8e8bdabeac41e066052d1ea3abcf68e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0529f2d163df9e2cd2ae8254dfb63806"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-20T12:54:26.883321073"
    },
    "sarscov2_sample_singleend_fqgz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true,
                            "sample_size": 50
                        },
                        "test.sampled_test_1.fastq.gz:md5,e5f44fafd7351c5abb9925a075132941"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0529f2d163df9e2cd2ae8254dfb63806"
                ],
                "reads": [
                    [
                        {
                            "id": "test",
                            "single_end": true,
                            "sample_size": 50
                        },
                        "test.sampled_test_1.fastq.gz:md5,e5f44fafd7351c5abb9925a075132941"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0529f2d163df9e2cd2ae8254dfb63806"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-06-20T12:54:21.119225479"
    }
}